fastchat/utils.py

```diff
@@ -418,41 +418,58 @@ def get_image_file_from_gcs(filename):
     return contents
 
 
-def pil_image_to_byte_array(pil_image, format="JPEG"):
-    img_byte_arr = BytesIO()
-    pil_image.save(img_byte_arr, format=format)
-    img_byte_arr = img_byte_arr.getvalue()
-    return img_byte_arr
+def resize_image_and_return_image_in_bytes(image, max_image_size_mb):
+    from PIL import Image
+    import math
 
+    image_bytes = BytesIO()
+    if not max_image_size_mb is None:
+        image.save(image_bytes, format="PNG")
+        target_size_bytes = max_image_size_mb * 1024 * 1024
+        current_size_bytes = image_bytes.tell()
 
-def convert_image_to_byte_array(image):
-    if type(image) == str:
-        with open(image, "rb") as image_data:
-            image_bytes = image_data.read()
+        if current_size_bytes > target_size_bytes:
+            resize_factor = (target_size_bytes / current_size_bytes) ** 0.5
+            new_width = math.floor(image.width * resize_factor)
+            new_height = math.floor(image.height * resize_factor)
+            resized_image = image.resize((new_width, new_height))
+
+            image_bytes = BytesIO()
+            resized_image.save(image_bytes, format="PNG")
+
+        image_bytes.seek(0)
     else:
-        image_bytes = pil_image_to_byte_array(
-            image, format="JPEG"
-        )  # Use 'PNG' or other formats as needed
+        image.save(image_bytes, format="PNG")
 
     return image_bytes
 
 
-def image_moderation_request(image, endpoint, api_key):
-    print(f"moderating image: {image}")
+def convert_image_to_byte_array(image):
+    from PIL import Image
 
-    headers = {"Content-Type": "image/jpeg", "Ocp-Apim-Subscription-Key": api_key}
+    if type(image) == str:
+        pil_image = Image.open(image).convert("RGB")
+        image_bytes = resize_image_and_return_image_in_bytes(
+            pil_image, max_image_size_mb=4
+        )
+    else:
+        image_bytes = resize_image_and_return_image_in_bytes(image, max_image_size_mb=4)
```

Comment:
why is there a default for max_image_size_mb? is this for the moderation API?

Reply:
good point! refactored it to remove the default since it's not immediately clear why in the code why it is defaulted to 4. But yes, you are correct - the default of 4 corresponds to the moderation API.